modules/gapi/perf/common/gapi_imgproc_perf_tests_inl.hpp

```diff
@@ -622,6 +622,54 @@ PERF_TEST_P_(CannyPerfTest, TestPerformance)
 
 //------------------------------------------------------------------------------
 
+PERF_TEST_P_(GoodFeaturesPerfTest, TestPerformance)
+{
+    compare_f cmpF;
+    MatType type;
+    int maxCorn = 0, blockSize = 0;
+    double qual = 0.0, minDist = 0.0, k = 0.0;
+    cv::Size sz;
+    bool useHarris = false;
+    cv::GCompileArgs compile_args;
+    std::tie(cmpF, type, sz, maxCorn, qual, minDist, blockSize, useHarris, k, compile_args) = GetParam();
+
+    initMatrixRandN(type, sz, CV_8UC1, false);
+
+    std::vector<cv::Point2f> out_vec_ocv, out_vec_gapi;
+    out_vec_ocv.reserve(maxCorn);
+    out_vec_gapi.reserve(maxCorn);
+
+    // OpenCV code /////////////////////////////////////////////////////////////
+    {
+        cv::goodFeaturesToTrack(in_mat1, out_vec_ocv, maxCorn, qual, minDist, cv::noArray(), blockSize, useHarris, k);
+        out_mat_ocv = cv::Mat(out_vec_ocv);
```

Comment:
Does it look OK to convert vector to Mat just not to implement compareVector() function?

Reply:
I think you need a custom comparison function here.

Reply:
implemented